### PR TITLE
add autoupdate debouncer

### PR DESCRIPTION
**Pull request by RasPhilCo**

### CODE REVIEW

File: src/plugins/plugin.ts [hunk 1]
```diff
@@ -170,7 +170,7 @@ export abstract class Plugin implements ICommandManager {
     const files = await deps.globby(['**/*.+(js|ts)', '!**/*.+(d.ts|test.ts|test.js)'], { nodir: true, cwd: d })
     const ids = files
       .map(path.parse)
-      .map(p => _.compact([...p.dir.split(path.sep), p.name === 'index' ? '' : p.name]).join(':'))
+      .map((p: any) => _.compact([...p.dir.split(path.sep), p.name === 'index' ? '' : p.name]).join(':'))
```
Comment (jdx) [inline src/plugins/plugin.ts:173]: is this needed?

File: src/commands/update.ts [hunk 2]
```diff
@@ -24,11 +25,18 @@ export default class Update extends Command {
   updater: Updater
 
   async run() {
+    this.updater = new Updater(this.config)
     // on manual run, also log to file
     if (!this.flags.autoupdate) {
```
Comment (jdx) [inline src/commands/update.ts]: I would flip this conditional, ! or else is always weird

File: src/updater.ts [hunk 3]
```diff
@@ -209,6 +215,14 @@ export class Updater {
     }
   }
 
+  private async touchLastRun(): Promise<void> {
+    try {
+      await deps.file.touch(this.lastrunfile)
```
Comment (jdx) [inline src/updater.ts]: we should probably just make it so touch does this

File: src/plugins/plugin.ts (hunk 1 above)
Comment (RasPhilCo) [reply src/plugins/plugin.ts:173]: my linter complains

File: src/updater.ts (hunk 3 above)
Comment (RasPhilCo) [reply src/updater.ts]: done

**Pull request merged**